### PR TITLE
Add reverse world blacklist system

### PR DESCRIPTION
**Pull request by LogGitDev**
Adds `reverse-world-blacklist-functionality: false` job config feature.

This system allows a job to function only in specific worlds. E.g.
```yaml
  #(Result: Prevents job in mainworld) - default
  reverse-world-blacklist-functionality: false
  world-blacklist:
  - mainworld
```
```yaml
  #(Result: Only allows job in mainworld)
  reverse-world-blacklist-functionality: true
  world-blacklist:
  - mainworld
```

### CODE REVIEW

File: src/main/java/com/gamingmesh/jobs/container/Job.java [hunk 1]
```diff
@@ -86,6 +86,7 @@
 
     private final List<String> fDescription = new ArrayList<>(), maxLevelCommands = new ArrayList<>();
     private List<String> worldBlacklist = new ArrayList<>();
+    private boolean reversedBlacklistFunctionality = false;
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/container/Job.java:89]: Rename to `reversedWorldBlacklist`

File: src/main/java/com/gamingmesh/jobs/container/Job.java [hunk 2]
```diff
@@ -643,17 +644,25 @@ public boolean isWorldBlackListed(Block block) {
...
+        if (worldBlacklist.isEmpty())
+            return isReversedBlacklistFunctionality();
 
-	if (block != null && worldBlacklist.contains(block.getWorld().getName()))
-	    return true;
+        if (block != null)
+            return worldBlacklist.contains(block.getWorld().getName()) != isReversedBlacklistFunctionality();
 
-	return ent != null && worldBlacklist.contains(ent.getWorld().getName());
+        return ent != null && worldBlacklist.contains(ent.getWorld().getName()) != isReversedBlacklistFunctionality();
```
Comment (montlikadani) [inline src/main/java/com/gamingmesh/jobs/container/Job.java:653]: Do not call directly the reversed method, instead use the field property.
```suggestion
        return ent != null && worldBlacklist.contains(ent.getWorld().getName()) != reversedWorldBlacklist;
```
(You need to apply the first review before committing this)

**Comment by LogGitDev** [comment]
@montlikadani Cheers, should all be clear now, let me know if any other issues things need to be changed.

**Comment by montlikadani** [comment]
You need to use the master branch instead of dev. You could force-move the target dev branch to master, but if you don't know then you can make a new PR.

**Pull request merged**